src/serial/frame_header_parser.cpp

```diff
@@ -0,0 +1,255 @@
+#include "adm/serial/frame_header_parser.hpp"
+#include "adm/private/xml_parser_helper.hpp"
+#include "adm/errors.hpp"
+#include "adm/private/rapidxml_utils.hpp"
+#include "rapidxml/rapidxml.hpp"
+#include "rapidxml/rapidxml_utils.hpp"
+#include <iostream>
+#include "adm/private/document_parser.hpp"
+#include "adm/serial/changed_ids.hpp"
+#include "adm/private/changed_id_traits.hpp"
+
+namespace adm {
+  namespace xml {
+    FrameHeaderParser::FrameHeaderParser(const std::string& filename,
+                                         ParserOptions options)
+        : FrameHeaderParser(rapidxml::file<>(filename.c_str()), options) {}
+
+    FrameHeaderParser::FrameHeaderParser(std::istream& stream,
+                                         ParserOptions options)
+        : FrameHeaderParser(rapidxml::file<>(stream), options) {}
+
+    FrameHeaderParser::FrameHeaderParser(rapidxml::file<> file,
+                                         ParserOptions options)
+        : file{std::move(file)}, options{std::move(options)} {}
+
+    /// Check if a option/flag is set
+    /**
+     * Checks if the option @a flag is set within @a options.
+     *
+     * This is equivalent to an bitwise AND followed by a conversion to bool,
+     * but should improve readability.
+     */
+    /*inline bool isSet(ParserOptions options, ParserOptions flag) {
+      return static_cast<bool>(options & flag);
+    }*/
```

Comment:
probably remove this (or move it to the right header)

Reply:
bc90561af33808864f4858406ce5b0cb95353b0e